src/component/MissionState/MissionState.tsx

```diff
@@ -1,17 +1,48 @@
-import React, {useEffect, useState} from "react"
+import CompoundedSpace from "antd/lib/space";
+import {useContext, useEffect, useState} from "react"
+import { FlightsContext } from "../../Providers/FlightsProvider/FlightsProvider";
 import "./MissionState.css"
 
-function MissionState() {
-    const [date, updateDate] = useState(0)
+const months = ["stycznia", "lutego", "marca", "kwietnia", "maja", "czerwca", "lipca", "sierpnia", "września", "października", "listopada", "grudnia"] as const;
+
+type TMissionState = "WAITING" | "LIVE" | "ENDED"
 
-    const dDayDate = new Date(2022, 4, 7, 12, 0)
-    const endDate = new Date(2022, 4, 7, 15, 30)
+interface IMissionDescription {
+    title: string,
+    description: string,
+    buttonText: string,
+}
 
+interface IMissionMessages {
+    waiting: IMissionDescription
+    live: IMissionDescription
+    ended: IMissionDescription
+}
 
-    const missionMessages = {
+type TDate = {
+    [key:string] : number | string
+}
+function MissionState() {
+    const [date, updateDate] = useState<TDate | 0>(0)
+    const [currentState, setCurrentState] = useState<TMissionState | "">("")
+    const { newest:currentFlight } = useContext(FlightsContext);
+    const dDayDate = new Date(currentFlight.date)
+
+    const [month, day, year] = [
+    dDayDate.getMonth(),
+    dDayDate.getDate(),
+    dDayDate.getFullYear(),
+    ];
+    const [hour, minutes, seconds] = [
+    dDayDate.getHours() < 10 ? "0"+dDayDate.getHours() : dDayDate.getHours(),
+    dDayDate.getMinutes() < 10 ? "0"+dDayDate.getMinutes() : dDayDate.getMinutes(),
+    dDayDate.getSeconds() < 10 ? "0"+dDayDate.getSeconds() : dDayDate.getSeconds(),
+    ];
```

Comment:
Imo this kind of calculation/transformation may be kept in separate files in directory like utils or smth to keep components more 'slim'.